lib/charms/ovn_charm.py

```diff
@@ -621,13 +641,18 @@ def __init__(self, **kwargs):
         """Allow augmenting behaviour on external factors."""
         super().__init__(**kwargs)
         if self.enable_openstack:
-            metadata_agent = 'networking-ovn-metadata-agent'
+            self.metadata_agent = 'networking-ovn-metadata-agent'
```

Comment:
If you move the definition of `self.nrpe_check_services` above this block you can append the metadata agent to it when OpenStack is enabled. Then you don't need to expose the `metadata_agent` variable as a class instance variable.